tests/provider: CodeArtifact service not supported in GovCloud

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

These tests fails on GovCloud:

```
TestAccAWSCodeArtifactAuthorizationTokenDataSource_basic
TestAccAWSCodeArtifactAuthorizationTokenDataSource_duration
TestAccAWSCodeArtifactAuthorizationTokenDataSource_owner
TestAccAWSCodeArtifactRepositoryEndpointDataSource_basic
TestAccAWSCodeArtifactRepositoryEndpointDataSource_owner
TestAccAWSCodeArtifactRepositoryPermissionsPolicy_basic
TestAccAWSCodeArtifactRepositoryPermissionsPolicy_disappears
TestAccAWSCodeArtifactRepositoryPermissionsPolicy_disappears_domain
TestAccAWSCodeArtifactRepositoryPermissionsPolicy_owner
```

With this error: `no such host`.

The CodeArtifact service does not exist on GovCloud and should be prechecked.

### New or Affected Resource(s)

<!--- Please list the new or affected resources and data sources. --->

* aws_codeartifact_domain_permissions_policy
* aws_codeartifact_domain
* aws_codeartifact_repository

### Terraform Configuration Files

Example config:

```hcl
resource "aws_codeartifact_repository" "test" {
  repository = "yakrepo"
  domain     = aws_codeartifact_domain.test.domain

  upstream {
    repository_name = aws_codeartifact_repository.upstream1.repository
  }

  upstream {
    repository_name = aws_codeartifact_repository.upstream2.repository
  }
}
```

### Debug Output

```
TestAccAWSCodeArtifactAuthorizationTokenDataSource_duration: data_source_aws_codeartifact_authorization_token_test.go:55: Step 1/1 error: Error running apply: 2020/10/23 05:13:08 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.10.0
        
        Error: error creating CodeArtifact Domain: RequestError: send request failed
        caused by: Post "https://codeartifact.us-gov-west-1.amazonaws.com/v1/domain?domain=tf-acc-test-9218336131831542759": dial tcp: lookup codeartifact.us-gov-west-1.amazonaws.com on 192.168.22.2:53: no such host
        
        
--- FAIL: TestAccAWSCodeArtifactAuthorizationTokenDataSource_duration (40.22s)
```

### References

* #0000

--- a/aws/data_source_aws_codeartifact_authorization_token_test.go
+++ b/aws/data_source_aws_codeartifact_authorization_token_test.go
@@ -13,7 +13,7 @@ func TestAccAWSCodeArtifactAuthorizationTokenDataSource_basic(t *testing.T) {
 	dataSourceName := "data.aws_codeartifact_authorization_token.test"
 
 	resource.Test(t, resource.TestCase{
-		PreCheck:  func() { testAccPreCheck(t) },
+		PreCheck:  func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck("codeartifact", t) },
 		Providers: testAccProviders,
 		Steps: []resource.TestStep{
 			{
@@ -33,7 +33,7 @@ func TestAccAWSCodeArtifactAuthorizationTokenDataSource_owner(t *testing.T) {
 	dataSourceName := "data.aws_codeartifact_authorization_token.test"
 
 	resource.Test(t, resource.TestCase{
-		PreCheck:  func() { testAccPreCheck(t) },
+		PreCheck:  func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck("codeartifact", t) },
 		Providers: testAccProviders,
 		Steps: []resource.TestStep{
 			{
@@ -53,7 +53,7 @@ func TestAccAWSCodeArtifactAuthorizationTokenDataSource_duration(t *testing.T) {
 	dataSourceName := "data.aws_codeartifact_authorization_token.test"
 
 	resource.Test(t, resource.TestCase{
-		PreCheck:  func() { testAccPreCheck(t) },
+		PreCheck:  func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck("codeartifact", t) },
 		Providers: testAccProviders,
 		Steps: []resource.TestStep{
 			{

--- a/aws/data_source_aws_codeartifact_repository_endpoint_test.go
+++ b/aws/data_source_aws_codeartifact_repository_endpoint_test.go
@@ -13,7 +13,7 @@ func TestAccAWSCodeArtifactRepositoryEndpointDataSource_basic(t *testing.T) {
 	dataSourceName := "data.aws_codeartifact_repository_endpoint.test"
 
 	resource.Test(t, resource.TestCase{
-		PreCheck:  func() { testAccPreCheck(t) },
+		PreCheck:  func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck("codeartifact", t) },
 		Providers: testAccProviders,
 		Steps: []resource.TestStep{
 			{
@@ -32,7 +32,7 @@ func TestAccAWSCodeArtifactRepositoryEndpointDataSource_owner(t *testing.T) {
 	dataSourceName := "data.aws_codeartifact_repository_endpoint.test"
 
 	resource.Test(t, resource.TestCase{
-		PreCheck:  func() { testAccPreCheck(t) },
+		PreCheck:  func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck("codeartifact", t) },
 		Providers: testAccProviders,
 		Steps: []resource.TestStep{
 			{

--- a/aws/resource_aws_codeartifact_repository_permissions_policy_test.go
+++ b/aws/resource_aws_codeartifact_repository_permissions_policy_test.go
@@ -17,7 +17,7 @@ func TestAccAWSCodeArtifactRepositoryPermissionsPolicy_basic(t *testing.T) {
 	resourceName := "aws_codeartifact_repository_permissions_policy.test"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheck(t) },
+		PreCheck:     func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck("codeartifact", t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckAWSCodeArtifactRepositoryPermissionsDestroy,
 		Steps: []resource.TestStep{
@@ -56,7 +56,7 @@ func TestAccAWSCodeArtifactRepositoryPermissionsPolicy_owner(t *testing.T) {
 	resourceName := "aws_codeartifact_repository_permissions_policy.test"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheck(t) },
+		PreCheck:     func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck("codeartifact", t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckAWSCodeArtifactRepositoryPermissionsDestroy,
 		Steps: []resource.TestStep{
@@ -84,7 +84,7 @@ func TestAccAWSCodeArtifactRepositoryPermissionsPolicy_disappears(t *testing.T) 
 	resourceName := "aws_codeartifact_repository_permissions_policy.test"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheck(t) },
+		PreCheck:     func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck("codeartifact", t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckAWSCodeArtifactRepositoryPermissionsDestroy,
 		Steps: []resource.TestStep{
@@ -105,7 +105,7 @@ func TestAccAWSCodeArtifactRepositoryPermissionsPolicy_disappears_domain(t *test
 	resourceName := "aws_codeartifact_repository_permissions_policy.test"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheck(t) },
+		PreCheck:     func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck("codeartifact", t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckAWSCodeArtifactRepositoryPermissionsDestroy,
 		Steps: []resource.TestStep{

